astropy/utils/data.py

```diff
@@ -1545,14 +1554,45 @@ def _keep_trying(timeout):
 # access to the shelve even for reading.
 @contextlib.contextmanager
 def _cache_lock(pkgname, need_write=False):
-    lockdir = os.path.join(_get_download_cache_locs(pkgname)[0], 'lock')
-    pidfn = os.path.join(lockdir, 'pid')
-    got_lock = False
-    try:
+    cache_dir = _get_download_cache_locs(pkgname)[0]
+    lock_name = 'lock'
+    lock_dir = os.path.join(cache_dir, lock_name)
+
+    def rmdir_fd(fd):
```

Comment:
To make it clearer that we are removing the lock dir inside the given `fd`, not removing `fd` itself.
```suggestion
    def rm_lockdir_fd(fd):
```